tests/unit/nupic/research/temporal_memory_test.py

```diff
@@ -59,368 +68,848 @@ def testInitInvalidParams(self):
-
-
-  def testActivateCorrectlyPredictiveCellsEmpty(self):
-    tm = self.tm
-
-    # No previous predictive cells, no active columns
-    prevPredictiveCells = set()
-    activeColumns      = set()
-    prevMatchingCells = set()
-
-    (activeCells,
-    winnerCells,
-    predictedColumns,
-    predictedInactiveCells) = tm.activateCorrectlyPredictiveCells(prevPredictiveCells,
-                                                                  prevMatchingCells,
-                                                                  activeColumns)
-
-    self.assertEqual(activeCells,      set())
-    self.assertEqual(winnerCells,      set())
-    self.assertEqual(predictedColumns, set())
-    self.assertEqual(predictedInactiveCells, set())
-
-    # No previous predictive cells, with active columns
-
-    prevPredictiveCells = set()
-    activeColumns = set([32, 47, 823])
-    prevMatchingCells = set()
-
-    (activeCells,
-    winnerCells,
-    predictedColumns,
-    predictedInactiveCells) = tm.activateCorrectlyPredictiveCells(prevPredictiveCells,
-                                                                  prevMatchingCells,
-                                                                  activeColumns)
-
-    self.assertEqual(activeCells,      set())
-    self.assertEqual(winnerCells,      set())
-    self.assertEqual(predictedColumns, set())
-    self.assertEqual(predictedInactiveCells, set())
-
-    # No active columns, with previously predictive cells
-
-    prevPredictiveCells = set([0, 237, 1026, 26337, 26339, 55536])
-    activeColumns = set()
-    prevMatchingCells = set()
-
-    (activeCells,
-    winnerCells,
-    predictedColumns,
-    predictedInactiveCells) = tm.activateCorrectlyPredictiveCells(prevPredictiveCells,
-                                                                  prevMatchingCells,
-                                                                  activeColumns)
-
-    self.assertEqual(activeCells,      set())
-    self.assertEqual(winnerCells,      set())
-    self.assertEqual(predictedColumns, set())
-    self.assertEqual(predictedInactiveCells, set())
-
-  def testActivateCorrectlyPredictiveCellsOrphan(self):
-    tm = self.tm
-    tm.predictedSegmentDecrement = 0.001
-    prevPredictiveCells = set([])
-    activeColumns = set([32, 47, 823])
-    prevMatchingCells = set([32, 47])
-
-    (activeCells,
-    winnerCells,
-    predictedColumns,
-    predictedInactiveCells) = tm.activateCorrectlyPredictiveCells(prevPredictiveCells,
-                                                                  prevMatchingCells,
-                                                                  activeColumns)
-
-    self.assertEqual(activeCells, set([]))
-    self.assertEqual(winnerCells, set([]))
-    self.assertEqual(predictedColumns, set([]))
-    self.assertEqual(predictedInactiveCells, set([32,47]))
-
-  def testBurstColumns(self):
+  def testActivateCorrectlyPredictedCells(self):
```

Comment:
I can't tell exactly from this diff but are you familiar with the distinction between "predictive" and "predicted" in this code? "predicted" is used to refer to predictions from the previous time step that you would compare or use in current activations. "predictive" is used to refer to cells in the current step that are predicted to be active in the next time step.

I don't like the terminology but we have to keep it consistent in this file or it will be impossible to understand. So make sure you use it accordingly.